### DISCUSSION TITLE
Filter with relationship - Table - List Records

### DISCUSSION THREAD

**Discussion by otavio-araujo** [filamentphp/filament#3023] [Help] [answered]
How can i use filters with relationships?

`Filter::make('liquidado_em')
                    ->form([
                        Forms\Components\DatePicker::make('liquidado_em'),
                        Forms\Components\DatePicker::make('liquidado_ate'),
                    ])
                    ->query(function (Builder $query, array $data): Builder {
                        return $query
                            ->join('contas_receber', 'transactions.id', 'contas_receber.id')
                            ->when(
                                $data['liquidado_em'],
                                fn (Builder $query, $date): Builder => $query->whereDate('contas_receber.liquidado_em', '>=', $date),
                            )
                            ->when(
                                $data['liquidado_ate'],
                                fn (Builder $query, $date): Builder => $query->whereDate('contas_receber.liquidado_em', '<=', $date), 
                            );
                    }), `

![image](https://user-images.githubusercontent.com/30505243/177046003-d87236a6-529d-4a57-b505-35937eda0938.png)

That's the way im trying to do, but im getting error:

![image](https://user-images.githubusercontent.com/30505243/177045920-7db73aaa-6eec-4bee-a2e3-a16276e6ea02.png)

**Comment by studiowizjo** [comment, marked as answer]
Instead of `whereDate()` use `whereHas()`. Check this in [Laravel documentation](https://laravel.com/docs/9.x/eloquent-relationships#querying-relationship-existence).

**Reply by otavio-araujo** [reply]
Thank you very much!
